backend/src/controllers/admin/deleteCourse.ts

```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { getClientCoursesByCourseId } from 'db/providers/clientCourseProvider';
+import { deleteCourseProvider } from 'db/providers/courseProvider';
+import CourseStatus from 'enums/coursesEnums';
+import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
+
+const deleteCourse = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const courseId = req.params.id;
+
+    const clientCourses = await getClientCoursesByCourseId(courseId);
```

Comment:
We can pass check by statuses inside DB provider with getting client courses. It will reduce amount of code in controller.
And it will allow us to throw an error in case no courses was returned, without an additional `find`

Reply:
done